Formula/talisman.rb

```diff
@@ -0,0 +1,35 @@
+class Talisman < Formula
+  desc "Detects secrets checked into git repository before commit or push"
+  homepage "https://thoughtworks.github.io/talisman/"
+  url "https://github.com/thoughtworks/talisman/archive/v0.4.3.tar.gz"
+  sha256 "3aa40773885cbdab4ef5aa5cdc2465879cfe2d61e91619be4a2fb034fb001ae1"
+
+  depends_on "go" => :build
+  depends_on "gox" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GO111MODULE"] = "on"
+    talisman_path = buildpath/"src/github.com/thoughtworks/talisman"
+    talisman_path.install buildpath.children
+
+    cd talisman_path do
+      arch = "amd64"
+      system "gox", "-osarch=darwin/#{arch}", "-ldflags=\"-X main.Version=v#{version}\"", "-output", bin/"talisman"
+      bin.install "global_install_scripts/setup_talisman.bash"
+      bin.install "global_install_scripts/remove_hooks.bash"
+    end
+  end
+
+  def caveats
+    <<~EOS
+      \x1b[33mTalisman has been setup in your machine.
+      To configure your git repositories to use Talisman, please follow instructions at
+      https://github.com/thoughtworks/talisman
+    EOS
+  end
+
+  test do
+    assert_equal "talisman Development Build\n", shell_output("#{bin}/talisman --v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.